### ISSUE TITLE
parent calls are (always) detected as calls outside the rule namespace

### ISSUE THREAD

**Issue by ricfio** [phparkitect/arkitect#201]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Library Version | 0.2.4 
| PHP version     | 8.0.12

#### Summary

false positive in parent::__construct call

#### Current behavior

parent::__construct is detected (false positive) as a call outside the rule namespace in configuration 
(also if the base class is inside a namespace matching with configuration rule namespace)

NOTE
could it miss a semantic analysis of the `parent` keyword needed to replace `parent' with the real className in the parsing tree ?

#### How to reproduce

Use this rule in configuration:
```
    $rules[] = Rule::allClasses()
        ->that(new ResideInOneOfTheseNamespaces('App\Domain'))
        ->should(new NotHaveDependencyOutsideNamespace('App\Domain'))
        ->because('we want protect our domain');
```

to check a class as this:
```
<?php

declare(strict_types=1);

namespace App\Domain\Asset;

use App\Domain\Collection;
use App\Domain\Common\CashFlow;
use App\Domain\Common\Profit;
use App\Domain\Investment\InvestmentState;
use App\Domain\Operation\OperationStat;
use DateTimeImmutable;

class AssetCollection extends Collection
{
    protected OperationStat $buy;
    protected OperationStat $sell;
    protected OperationStat $current;

    /** @psalm-immutable */
    protected ?Profit $profit = null;

    public function __construct(array $assets = [])
    {
        parent::__construct($assets);
...
```

#### Expected behavior

No checking error for parent calls if the base class is inside a namespace matching with configuration rule namespace)

**Comment by micheleorselli** [comment]
Should that be fixed with #197?

**Comment by ricfio** [comment]
Yes, I suppose so

**Comment by AlessandroMinoccheri** [comment]
Fixed by #197
